tf2_ros/src/buffer.cpp

```diff
@@ -242,10 +242,12 @@ Buffer::waitForTransform(const std::string& target_frame, const std::string& sou
     // Immediately transformable
     geometry_msgs::msg::TransformStamped msg_stamped = lookupTransform(target_frame, source_frame, time);
     promise->set_value(msg_stamped);
+    removeTransformableCallback(cb_handle);
   } else if (0xffffffffffffffffULL == handle) {
     // Never transformable
     promise->set_exception(std::make_exception_ptr<tf2::LookupException>(
           "Failed to transform from " + source_frame + " to " + target_frame));
+    removeTransformableCallback(cb_handle);
```

Comment:
These look good to me.  However, I think there is also one more case missing, in the `timerCallback`.  As I understand it, if the timer expires before the future is completed, then the timer will go and call `cancelTransformableRequest`.  At that point, it *also* needs to call `removeTransformableCallback`, otherwise we will have a leak there, I believe.

Reply:
I agree. I believe we only need to call `removeTransformavleCallback`, since it iterates over the transformable requests and removes all associated requests.